specification/apimanagement/developer-services-api/authorized/resources/identities.tsp

```diff
@@ -0,0 +1,57 @@
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+import "../../models/common.tsp";
+import "../../models/identity.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.Core;
+using DeveloperServicesApi.Models;
+using DeveloperServicesAuthorizedApi.Common;
+
+namespace DeveloperServicesAuthorizedApi.Resources.Confirmations;
+
+interface Identities {
+  @doc("Identity of current user.")
+  getIdentity is ResourceRead<IdentityResource>;
+
+  @doc("List a collection of user identities.")
+  listIdentities is ResourceList<UserIdentityResource>;
+
+  @route("/users/{userId}/identities/{provider}/{identity}")
+  @doc("Creates or updates the given user identity.")
+  @put
+  CreateOrUpdateUserIdentityProvider(
```

Comment:
This should be implemented with an Azure Core operation template, perhaps `ResourceCreateOrReplace`.